source/MaterialXCore/Document.cpp

```diff
@@ -401,6 +402,32 @@ void Document::invalidateCache()
     _cache->valid = false;
 }
 
+void Document::normalizeValueStrings(unsigned int numericPrecision)
+{
+    // For now this will only normalize numeric strings
+    StringSet numericType = { "float", "vector2", "vector3", "vector4", "color3", "color4" };
```

Comment:
I'm not sure if it's legal or not, but do we also want to consider removing unnecessary leading zeros from integer values? 

And we may also want to include matrix types here too.

Reply:
To be safe, I've only removed leading and trailing spaces for integers.